src/modules/gradle.rs

```diff
@@ -0,0 +1,243 @@
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use crate::{
+    config::ModuleConfig,
+    configs::gradle::{GradleConfig, GradleVersionStrategy},
+    context::Context,
+    formatter::{StringFormatter, VersionFormatter},
+    module::Module,
+};
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("gradle");
+    let config = GradleConfig::try_load(module.config);
+    let is_gradle_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_gradle_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let gradle_version = match config.strategy {
+                        GradleVersionStrategy::Executable => parse_gradle_version_from_stdout(
+                            &context
+                                .exec_cmds_return_first(vec![
+                                    vec!["./gradlew", "--version"],
```

Comment:
I would avoid executing files from the current working directory unless explicitly configured to do so.

Reply:
99.9% of gradle projects use the gradle wrapper, so most of the users don't even have gradle installed, also the versions can differ from project to project, so displaying the globally installed version would be wrong.

Reply:
I'm just concerned about running executables from potentially untrusted directories.

Reply:
Oh yeah, I didn't think of that.